### DISCUSSION TITLE
MobileSAM (Mobile Segment Anything Model) - Ultralytics YOLOv8 Docs

### DISCUSSION THREAD

**Discussion by giscus** [ultralytics/ultralytics#4300] [Docs]
# MobileSAM (Mobile Segment Anything Model) - Ultralytics YOLOv8 Docs

Learn more about MobileSAM, its implementation, comparison with the original SAM, and how to download and test it in the Ultralytics framework. Improve your mobile applications today.

https://docs.ultralytics.com/models/mobile-sam/

**Comment by Muhammad-Kaleem-Ullah** [comment]
Unable to segment the image properly without prompting.

**Reply by glenn-jocher** [reply]
@Muhammad-Kaleem-Ullah i'm sorry to hear you're having trouble with image segmentation. To ensure proper segmentation without prompting, please make sure you're following these steps:

1. Use a pre-trained segmentation model or train your own model with a dataset that closely matches your use case.
2. Verify that your dataset is correctly annotated for segmentation tasks.
3. Check that the input images are preprocessed correctly as required by the model.
4. Ensure that the model is being used in the correct mode for segmentation.

If you're still facing issues, consider reviewing the model's documentation on segmentation and double-checking your setup. If the problem persists, please provide more details about your dataset, model configuration, and the steps you've taken so far, and we'll do our best to assist you further. 🛠️

**Comment by Muhammad-Kaleem-Ullah** [comment]
How can we use it to apply `segment_everything()` like we do in `FastSAM`?

**Reply by glenn-jocher** [reply]
@Muhammad-Kaleem-Ullah to apply segmentation similar to `segment_everything()` in MobileSAM, you would follow a process akin to what you're familiar with in FastSAM. First, ensure you have a trained MobileSAM model. Then, you can use the `Predict` mode to perform segmentation on your images or videos.

Here's a high-level overview of the steps you would take:

1. Load your trained MobileSAM model.
2. Prepare your input data (images or videos).
3. Use the `Predict` mode to run the model on your data.
4. The model will output segmented images or video frames.

Remember, the `Predict` mode is designed to be straightforward to use, allowing you to easily apply the model to new data for tasks like segmentation. For detailed instructions and options available for the `Predict` mode, you can refer to the Ultralytics Docs. 📚

If you need further guidance on using MobileSAM for segmentation, the documentation provides comprehensive information on the various modes and their usage.
